apps/src/screens/SwipeBackAnimation.tsx

```diff
@@ -27,12 +28,36 @@ interface ScreenBProps {
   navigation: NativeStackNavigationProp<StackParamList, 'ScreenB'>;
 }
 
-const ScreenB = ({ navigation }: ScreenBProps): React.JSX.Element => (
-  <View style={{ ...styles.container, backgroundColor: 'thistle' }}>
-    <Button title="Go ScreenC" onPress={() => navigation.navigate('ScreenC')} />
-    <Button title="Go back" onPress={() => navigation.goBack()} />
-  </View>
-);
+const ScreenB = ({ navigation }: ScreenBProps): React.JSX.Element => {
+  const [gestureType, setGestureType] = React.useState<NonNullable<NativeStackNavigationOptions['gestureType']>>('twoDimensionalSwipe');
+
+  React.useEffect(() => {
+    navigation.setOptions({
+      gestureType,
+    });
+  }, [gestureType]);
```

Comment:
Why do we move this into `setOptions` instead of leaving it set directly at screen level? 

Another thing is that me might use to opportunity to use `useLayoutEffect` here. I know it does not work as expected, but once it does we'll have this in place already.

Reply:
I updated the PR description with a preview. I extended the example with the ability to switch between all possible gestures.

From what I researched for SearchBar, `useLayoutEffect` works as expected, and it won't solve our problems here.